### PR TITLE
Parameterize MCMS configuration

### PR DESCRIPTION
**Pull request by connorwstein**
- Support multiple parameterized MCMS deployments
- Standardize on mutating address book inline
- Extract fee token deployment separately, so DeployCCIPContracts can be used for existing chains where link/weth exists

### CODE REVIEW

File: integration-tests/deployment/ccip/deploy.go [hunk 1]
```diff
@@ -55,18 +53,8 @@ var (
 	// Note test router maps to a regular router contract.
 	TestRouter   deployment.ContractType = "TestRouter"
 	CCIPReceiver deployment.ContractType = "CCIPReceiver"
-
-	TestXXXMCMSSigner *ecdsa.PrivateKey
```
Comment (connorwstein) [inline integration-tests/deployment/ccip/deploy.go:59]: pulled this into propose.go alongside other MCMS related actions

File: integration-tests/deployment/ccip/deploy.go [hunk 2]
```diff
@@ -291,53 +259,91 @@ func DeployChainContracts(
...
 
-	timelock, err := deployContract(e.Logger, chain, ab,
+	timelock, err := deployContract(lggr, chain, ab,
 		func(chain deployment.Chain) ContractDeploy[*owner_helpers.RBACTimelock] {
 			timelock, tx, cc, err2 := owner_helpers.DeployRBACTimelock(
 				chain.DeployerKey,
 				chain.Client,
 				big.NewInt(0), // minDelay
-				mcm.Address,
-				// TODO: Actual MCM groups need to be parameterized.
```
Comment (connorwstein) [inline integration-tests/deployment/ccip/deploy.go:305]: the main aim of the ticket

File: integration-tests/deployment/ccip/deploy.go [hunk 3]
```diff
@@ -291,53 +259,91 @@ func DeployChainContracts(
...
+		feeTokenContractsByChain[chain.Selector] = feeTokenContracts
 	}
-	e.Logger.Infow("deployed rmnProxy", "addr", rmnProxy.Address)
+	return feeTokenContractsByChain, nil
+}
 
-	// TODO: Need general configuration for using pre-existing weth9
-	// link tokens.
-	weth9, err := deployContract(e.Logger, chain, ab,
+// DeployFeeTokens deploys link and weth9. This is _usually_ for test environments only,
```
Comment (connorwstein) [inline integration-tests/deployment/ccip/deploy.go:347]: preping for staging deploy where fee tokens are pre-existing

File: integration-tests/deployment/ccip/deploy_test.go [hunk 4]
```diff
@@ -21,32 +22,25 @@ func TestDeployCCIPContracts(t *testing.T) {
...
-	require.NoError(t, err)
-
-	feedChain := e.AllChainSelectors()[FeedChainIndex]
-	feedAddresses, _, err := DeployFeeds(lggr, e.Chains[feedChain])
-	require.NoError(t, err)
-
-	// Merge the feed addresses into the address book.
-	require.NoError(t, addressBook.Merge(feedAddresses))
+	ab := deployment.NewMemoryAddressBook()
+	homeChainSel, feedChainSel := allocateCCIPChainSelectors(e.Chains)
```
Comment (connorwstein) [inline integration-tests/deployment/ccip/deploy_test.go:26]: extracted this allocateCCIPChainSelector helper since same code exists twice

**Review by cmalec — APPROVED** [review]

**Review by ogtownsend — APPROVED** [review]

**Review by AnieeG — APPROVED** [review]

**Pull request merged**